pip/commands/list.py

```diff
@@ -91,6 +91,13 @@ def __init__(self, *args, **kw):
             help="Align package names and versions into vertical columns.",
         )
 
+        cmd_opts.add_option(
+            '--leaves',
+            action='store_true',
+            dest='leaves',
+            help="List packages that are not dependencies of installed packages.",
```

Comment:
This doesn't agree with the code, which reports all packages that have no dependencies themselves.

I'm not actually sure what the benefit is of only reporting packages with no dependencies. The original issue seemed to want the set of packages that, if you installed them, would replicate everything you had - but that's packages which aren't depended on by anything else, rather than what's implemented here.